lib/ansible/modules/database/mongodb/mongodb_role.py

```diff
@@ -0,0 +1,452 @@
+        default: "CERT_REQUIRED"
+        choices: ["CERT_REQUIRED", "CERT_OPTIONAL", "CERT_NONE"]
+    roles:
+        description:
+            - An array of roles from which this role inherits
+    authentication_restrictions:
+        description:
+            - The authentication restrictions the server enforces on the role
+    state:
+        description:
+            - The database role state
+        default: present
+        choices: [ "present", "absent" ]
+
+notes:
+    - Requires the pymongo Python package on the remote host, version 2.4.2+. This
+      can be installed using pip or the OS package manager. @see http://api.mongodb.org/python/current/installation.html
+    - Authentication Restrictions are not present in rolesInfo result up to MongoDB 4.0, so are not supported here in such versions.
+    - This module is loosely based on mongodb_user by Elliott Foster (@elliotttf) and Julien Thebault (@lujeni).
+requirements: [ "pymongo" ]
+author:
+    - "Jakub Kramarz (@jkramarz)"
+'''
+
+EXAMPLES = '''
+# Create 'overlord' role with inherited 'readWrite' role privileges in database 'test'.
+- mongodb_role:
+    database: test
+    name: overlord
+    roles:
+        - readWrite
+    state: present
+
+# Create role 'producer' in database 'test' with 'insert' privilege to 'whatever' collection in database 'test' and 'readWrite' access in database 'private'.
+- mongodb_role:
+    database: test
+    name: producer
+    roles:
+        - db: private
+          role: readWrite
+    privileges:
+        - resource:
+            db: test
+            collection: whatever
+          actions:
+            - insert
+    state: present
+
+# Remove role 'kaktus' from database 'test' via SSL (MongoDB must be compiled with the SSL option and configured properly)
+- mongodb_role:
+    database: test
+    name: kaktus
+    state: absent
+    ssl: True
+'''
+
+RETURN = '''
+role:
+    description: The name of the role to add or remove.
+    returned: success
+    type: string
+'''
+
+import os
+import ssl as ssl_lib
+import traceback
+from distutils.version import LooseVersion
+
+try:
+    import yaml
+    HAS_YAML = True
+except ImportError:
+    HAS_YAML = False
+
+try:
+    from pymongo.errors import ConnectionFailure
+    from pymongo.errors import OperationFailure
+    from pymongo import version as PyMongoVersion
+    from pymongo import MongoClient
+except ImportError:
+    try:  # for older PyMongo 2.2
+        from pymongo import Connection as MongoClient
+    except ImportError:
+        HAS_PYMONGO = False
+    else:
+        HAS_PYMONGO = True
+else:
+    HAS_PYMONGO = True
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.six.moves import configparser
+from ansible.module_utils._text import to_native
+
+
+# =========================================
+# MongoDB module specific support methods.
+#
+
+def check_compatibility(module, client):
```

Comment:
There's a hanging PR implementing it in a nicer way: https://github.com/ansible/ansible/pull/44110/files#diff-d4702ee9f9f2977f8c82cb5b4e8b0e26R221. You may want to try borrowing some ideas from there.